internal/engine/build_and_deployer.go

```diff
@@ -133,3 +136,17 @@ func DefaultBuildOrder(lubad *LiveUpdateBuildAndDeployer, ibad *ImageBuildAndDep
 
 	return BuildOrder{lubad, dcbad, ibad, ltbad}
 }
+
+func sendBuildCompleteTiming(ctx context.Context, startTime time.Time, updateType string, err error) {
+	hasErr := fmt.Sprintf("%t", err != nil)
+
+	analytics.Get(ctx).Timer("update", time.Since(startTime), map[string]string{
```

Comment:
Sorry I think you're misunderstanding my objection to this change. Let me try again, zooming out a bit to give you a bit of context

Most metrics frameworks have strong semantic conventions around what how metrics should be named and tagged. They have long docs on this. They don't do this because metrics people like writing long docs! They do it because it helps a ton when we need to do long-term analysis to have agreed upon conventions. 

We don't want the convention to be "whatever is easiest for the next query we want to write", since the format has big long-term consequences

What I'm asking for here is not "let's add more formats for continuity". I'm asking you to do a a small amount of upfront work to explain:

- why is this new format better?
- what existing conventions are you basing it on?
- if you're introducing a new convention, have you considered the trade-offs?

Here's a typical doc in this space:
https://github.com/opentracing/specification/blob/master/specification.md#start-a-new-span
(Though by no means the only doc)

If possible, I would rather keep the same "build.[type]" convention for now. I think the general "one metric name per function" is a good one. And as we move towards opencensus metrics, we should consider more thoroughly what our tag and name architecture should look like

Reply:
I thought I spoke to my reasoning behind this change above:
> I like [this new way of reporting the metric] because it nests EVERYTHING under `update` so you can query for all `update` events OR check the type of update by filtering on tags (instead of looking for `name == build.image || name == build.container || ...`).
To give more context, I had a talk with Victor yesterday in which we wanted to know whether ANY builds had been done (not caring about specific types) and it was doable-but-kind-of-a-pain to pull metrics for all the specific builds by name, plus it'd be be annoying to keep queries up to date if we add new build types.

Without knowing the specifics of how we retrieve data from the DB, I would ASSUME that this way of storing data (as an "update" event with a "type" tag) gives all the functionality of the previous system with additional ease of looking at all update events at once. (Though there may be e.g. performance nuances of filtering by tags that i'm missing.)

But sounds like either you find that reasoning unpersuasive or incomplete, or that the burden of proof for changing this is higher than I imagined, which is fine. I can just stick to the existing schema for now and we can discuss this more later/if the problems I anticipate keep being annoying.

Reply:
ya, this might be where we're talking past each other - "this was easiest for the query i was writing right now" isn't really the level of analysis I was looking for, and we'll leave it at that.

Reply:
ouch, I don't think that summarizes what I was saying. it sounds like you think I'm ignoring your request/criteria, which I'm not, I'm just not clear on what you're looking for or what would be a sufficiently compelling argument.